src/main/java/com/microsoft/cse/helium/app/dao/MoviesDao.java

```diff
@@ -0,0 +1,48 @@
+package com.microsoft.cse.helium.app.dao;
+
+import static com.microsoft.azure.spring.data.cosmosdb.exception.CosmosDBExceptionUtils.findAPIExceptionHandler;
+
+import com.azure.data.cosmos.CosmosClient;
+import com.microsoft.cse.helium.app.models.Movie;
+import com.microsoft.cse.helium.app.services.configuration.IConfigurationService;
+import com.microsoft.cse.helium.app.utils.CommonUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import reactor.core.publisher.Mono;
+
+@Service
+public class MoviesDao extends BaseCosmosDbDao {
+
+  @Autowired CommonUtils utils;
+  /*
+  private static String movieSelect =
+      "select m.id, m.partitionKey, m.movieId, m.type, m.textSearch, m.title, m.year, m.runtime,"
+          + "m.rating, m.votes, m.totalScore, m.genres, m.roles from m where m.type = 'Movie'  ";
+
+  private static String movieContains = "and contains(m.textSearch, \"%s\") ";
+  private static String movieOrderBy = " order by m.textSearch ";
+  private static String movieOffset = " offset %d limit %d ";*/
+
+  /** MoviesDao. */
+  public MoviesDao(IConfigurationService configService) {
+    super(configService);
+  }
+
+  /** getMovieByIdSingleRead. */
+  public Mono<Movie> getMovieById(String movieId) {
+
+    Mono<Movie> movie =
+        this.context
+            .getBean(CosmosClient.class)
+            .getDatabase(this.cosmosDatabase)
+            .getContainer(this.cosmosContainer)
+            .getItem(movieId, utils.getPartitionKey(movieId))
+            .read()
+            .flatMap(
+                cosmosItemResponse -> {
+                  return Mono.justOrEmpty(cosmosItemResponse.properties().toObject(Movie.class));
+                })
+            .onErrorResume(throwable -> findAPIExceptionHandler("Failed to find item", throwable));
```

Comment:
May we change the error message to "Failed to find movie with Id: " + movieId?

Reply:
Done!